Sources/MongoSwift/BSON/BSONDocumentIterator.swift

```diff
@@ -177,8 +177,8 @@ public class BSONDocumentIterator: IteratorProtocol {
      * Overwrites the current value of this `BSONDocumentIterator` with the supplied value.
      *
      * - Throws:
-     *   - `MongoError.InternalError` if the new value is an `Int` and cannot be written to BSON.
-     *   - `MongoError.LogicError` if the new value is a `BSONDecimal128` or `BSONObjectID` and is improperly formatted.
+     *   - `BSONError.InternalError` if the new value is an `Int` and cannot be written to BSON.
```

Comment:
I think this and the next error listed here can also go away, not relevant for same reasons as above